pkg/controllers/oauthclientscontroller/oauthclientscontroller.go

```diff
@@ -159,8 +159,12 @@ func randomBits(bits uint) []byte {
 }
 
 func ensureOAuthClient(ctx context.Context, oauthClients oauthclient.OAuthClientInterface, client oauthv1.OAuthClient) error {
-	_, err := oauthClients.Create(ctx, &client, metav1.CreateOptions{})
-	if err == nil || !apierrors.IsAlreadyExists(err) {
+	_, err := oauthClients.Get(ctx, client.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
Nit: maybe unnest the nested if? You can check whether `apierrors.IsNotFound()` regardless of whether `err` is `nil`.